src/services/TechRecordsService.ts

```diff
@@ -232,27 +232,72 @@ class TechRecordsService {
           // systemNumber search should return a unique record
           throw new HTTPError(500, ERRORS.NO_UNIQUE_RECORD);
         }
-        const uniqueRecord = data[0];
+        const uniqueRecord = data[0];// call payload
         const oldTechRec = this.getTechRecordToArchive(uniqueRecord, statusCode);
-        const newRecord: any = _.cloneDeep(oldTechRec);
-        oldTechRec.statusCode = STATUS.ARCHIVED;
-        _.mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
-        this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
-        populateFields(newRecord);
-        uniqueRecord.techRecord.push(newRecord);
-        return uniqueRecord;
+        if(!newStatusCode) {
+          const newRecord: ITechRecord = cloneDeep(oldTechRec);
+          mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
+          this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
+          populateFields(newRecord);
+          uniqueRecord.techRecord.push(newRecord);
+          return uniqueRecord;
+        } else {
+
+          if(newStatusCode === STATUS.ARCHIVED) {
+            console.log("Should get here");
+            this.updateAtributesWhenNewStatusIsArchived(oldTechRec,msUserDetails);
+            console.log("New Status code: ",uniqueRecord.techRecord[0].statusCode);
+            return uniqueRecord;
+
+          } else if(newStatusCode === STATUS.CURRENT && oldTechRec.statusCode === STATUS.PROVISIONAL) {
+            if(uniqueRecord.techRecord.filter((techRecords: any) => {
```

Comment:
Something doesn't seem right here. I think what you want to do here is check if the vehicle already has a current record. If that's the case, then you should also archive the existing current, beside the provisional, like it says in AC3.

I'd suggest to move the filter outside of the IF and retrieve the current record into a constant currentRecord. Check if currentRecord exists, if it does -> archive the current as well.